shell/gpu/gpu_surface_metal.mm

```diff
@@ -93,65 +95,52 @@
-  if (!metal_texture) {
-    FML_LOG(ERROR) << "Could not acquire metal texture from drawable.";
-    return nullptr;
-  }
-
-  GrMtlTextureInfo metal_texture_info;
-  metal_texture_info.fTexture.reset(SkCFSafeRetain(metal_texture.get()));
-
-  GrBackendRenderTarget metal_render_target(bounds.width * scale,   // width
-                                            bounds.height * scale,  // height
-                                            1,                      // sample count
-                                            metal_texture_info      // metal texture info
+  auto surface = SkSurface::MakeFromCAMetalLayer(context_.get(),            // context
+                                                 layer_.get(),              // layer
+                                                 kTopLeft_GrSurfaceOrigin,  // origin
+                                                 1,                         // sample count
+                                                 kBGRA_8888_SkColorType,    // color type
+                                                 nullptr,                   // colorspace
+                                                 nullptr,                   // surface properties
+                                                 &next_drawable_  // drawable (transfer out)
   );
 
-  auto command_buffer =
-      fml::scoped_nsprotocol<id<MTLCommandBuffer>>([[command_queue_.get() commandBuffer] retain]);
-
-  SkSurface::RenderTargetReleaseProc release_proc = [](SkSurface::ReleaseContext context) {
-    [reinterpret_cast<id>(context) release];
-  };
-
-  auto surface =
-      SkSurface::MakeFromBackendRenderTarget(context_.get(),            // context
-                                             metal_render_target,       // backend render target
-                                             kTopLeft_GrSurfaceOrigin,  // origin
-                                             kBGRA_8888_SkColorType,    // color type
-                                             nullptr,                   // colorspace
-                                             nullptr,                   // surface properties
-                                             release_proc,              // release proc
-                                             metal_texture.release()    // release context (texture)
-      );
-
   if (!surface) {
     FML_LOG(ERROR) << "Could not create the SkSurface from the metal texture.";
     return nullptr;
   }
 
-  bool hasExternalViewEmbedder = delegate_->GetExternalViewEmbedder() != nullptr;
-
   // External views need to present with transaction. When presenting with
   // transaction, we have to block, otherwise we risk presenting the drawable
   // after the CATransaction has completed.
   // See:
   // https://developer.apple.com/documentation/quartzcore/cametallayer/1478157-presentswithtransaction
   // TODO(dnfield): only do this if transactions are actually being used.
   // https://github.com/flutter/flutter/issues/24133
```

Comment:
nit: maybe we should move this comment down to line 143 now, since we're no longer getting whether it's an external view embedder here.

Reply:
Done.